app/src/ui/branches/branch-list.tsx

```diff
@@ -80,6 +82,15 @@ interface IBranchListProps {
 
   /** Callback to fire when the filter text is changed */
   readonly onFilterTextChanged: (filterText: string) => void
+
+  /** Should it let users create a new branch? */
+  readonly allowCreateNewBranch: boolean
```

Comment:
Could we stick to the `canXYZ` convention on props - e.g. `canSelectRow`, `canCommit` - here?